Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -1216,26 +1217,39 @@ def error_parser(self, error: requests.Response) -> str:
         try:
             response = error.json()
             demisto.error(str(response))
-            inner_error = response.get('error', {})
-            error_codes = response.get("error_codes", [""])
-            err_desc = response.get('error_description', '')
+            err_str = self.extract_microsoft_error(response)
+            return err_str or error.text
+        except Exception as e:
+            demisto.error(f'Failed parsing JSON error response - {e}')
 
-            if isinstance(inner_error, dict):
-                err_str = f"{inner_error.get('code')}: {inner_error.get('message')}"
-            else:
-                err_str = inner_error
-                re_search = re.search(REGEX_SEARCH_ERROR_DESC, err_desc)
-                err_str += f". \n{re_search['desc']}" if re_search else ""
+    def extract_microsoft_error(self, response: dict) -> str | None:
```

Comment:
why not to return empty string and then on line 906 you will need only to do:
`microsoft_response = self.extract_microsoft_error(json.loads(search_microsoft_response.group()))`

Reply:
but on line 906, I am checking that the regex result (_search_microsoft_response_) is not None before calling the function..
```
        _search_microsoft_response_ = re.search(r'{.*}', oproxy_response.text)
            microsoft_response = self.extract_microsoft_error(json.loads(search_microsoft_response.group())) \
                if _search_microsoft_response_ else ""
```
I can change the function to receive None|dict and then if None return ""